packages/styled-components/src/models/StyleSheetManager.js

```diff
@@ -17,40 +18,56 @@ export const StyleSheetConsumer = StyleSheetContext.Consumer;
 export const masterSheet: StyleSheet = new StyleSheet(false);
 
 export function useStyleSheet(): StyleSheet {
-  const sheet = useContext(StyleSheetContext);
-  return sheet !== undefined ? sheet : masterSheet;
+  return useContext(StyleSheetContext) || masterSheet;
 }
 
-function useStyleSheetProvider(sheet?: StyleSheet, target?: HTMLElement) {
-  return useMemo(
+export default function StyleSheetManager(props: Props) {
+  const styleSheet = useMemo(
     () => {
-      if (sheet) {
-        return sheet;
-      } else if (target) {
-        return new StyleSheet(false, target);
+      let sheet;
+
+      if (props.sheet) {
+        // eslint-disable-next-line prefer-destructuring
+        sheet = props.sheet;
+      } else if (props.target) {
+        sheet = new StyleSheet(false, props.target);
       } else {
-        return masterSheet;
+        sheet = masterSheet;
       }
-    },
-    [sheet, target]
-  );
-}
 
-export default function StyleSheetManager(props: Props) {
-  const sheet = useStyleSheetProvider(props.sheet, props.target);
+      if (props.disableCSSOMInjection) {
+        if (sheet === masterSheet) {
+          sheet = new StyleSheet(false);
+          sheet.useCSSOM = false;
+        } else {
+          sheet.useCSSOM = false;
+        }
+      }
+
+      if (props.stylisOptions || props.stylusPlugins) {
+        sheet.stringifier = createStylisInstance(props.stylisOptions, props.stylusPlugins);
+      }
 
-  if (props.stylisOptions || props.stylusPlugins) {
-    sheet.stringifier = createStylisInstance(props.stylisOptions, props.stylusPlugins);
-  }
+      return sheet;
+    },
+    [
+      props.disableCSSOMInjection,
+      props.sheet,
+      props.stylisOptions,
+      props.stylusPlugins,
```

Comment:
That's a typo here and the above and this are likely to cause issues here if people forget to keep their references equal. Given the severity of some people accidentally making this mistake, should we just make these constant on mount?

Reply:
Thanks for catching this 🤦‍♂

Reply:
@kitten done in the latest commit 👍